pkgs/applications/science/math/gap/default.nix

```diff
@@ -104,34 +118,34 @@ stdenv.mkDerivation rec {
     # like the defaults the Makefile, but use gap from PATH instead of the
     # one from builddir
     installCheckFlagsArray+=(
-      "TESTGAP=gap --quitonbreak -b -m 100m -o 1g -q -x 80 -r -A"
-      "TESTGAPauto=gap --quitonbreak -b -m 100m -o 1g -q -x 80 -r"
+      "TESTGAPcore=gap --quitonbreak -b -q -r"
+      "TESTGAPauto=gap --quitonbreak -b -q -r -m 100m -o 1g -x 80"
+      "TESTGAP=gap --quitonbreak -b -q -r -m 100m -o 1g -x 80 -A"
     )
   '';
 
   postBuild = ''
     pushd pkg
-    bash ../bin/BuildPackages.sh
+    bash ../bin/BuildPackages.sh --strict
     popd
   '';
 
-  installTargets = [
-    "install-libgap"
-    "install-headers"
-  ];
-
-  # full `make install` is not yet implemented, just for libgap and headers
   postInstall = ''
     # Install config.h, which is not currently handled by `make install-headers`
-    cp gen/config.h "$out/include/gap"
+    cp build/config.h "$out/include/gap"
 
-    mkdir -p "$out/bin" "$out/share/gap/"
+    # make install creates an empty pkg dir. the other directories are tested
+    # by make check
+    rm -r "$out/share/gap/pkg"
+    cp -ar pkg tst "$out/share/gap"
 
-    echo "Copying files to target directory"
-    cp -ar . "$out/share/gap/build-dir"
+    # install-bin creates a symlink pointing to /build/gap-X.Y.Z to support
+    # packages which include "src/compiled.h". we just remove the
+    # symlink, but if anyone complains we can copy the full build dir.
```

Comment:
We previously copied the full build directory to the output. Now that I've migrated to using the new install target, this is apparently only required for old kernel extensions, so I am not copying it any more. Tests still pass, please let me know if this is wrong.

Reply:
A bunch of packages still do `#include "src/compiled.h"`. They won't compile when built against an installed GAP unless this symlink is present (or of course alternatively a copy of the files in a `src` subdir). I've been working on getting packages to resolve this, but I am not sure it'll be ready in time for 4.12.1

Reply:
The `install-headers` symlink was fine, I just got confused by the fact that `make install` errored out right after running `ln`. The problem ended up being the (now fixed) dev docs issue, but I forgot to revisit this afterwards. So nothing needed to be done here. Sorry for the noise!